File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        if nums.count(0)>1:
            return [0]*len(nums)
        pro = 1
        zero = False
        for i in nums:
            if i == 0:
                zero = True
            else:
                pro *= i
        res = [0]*len(nums)
        for i in range(len(nums)):
            if nums[i] == 0:
                res[i] = pro
            elif nums[i] != 0 and not zero:
                res[i] = pro//nums[i]
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> mp;
        for(auto i: nums){
            mp[i] += 1;
            if (mp[i]>1)
                return true;
        }
        return false;
        
    }
};